client/src/pages/hospitals/HospitalModel.jsx

```diff
@@ -0,0 +1,141 @@
+import { Modal, TextInput, Group, Button } from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { notifications } from '@mantine/notifications';
+import PropTypes from 'prop-types';
+
+/**
+ * Invite Modal
+ *  @param {object} props - The prop for the component
+ *  @param {boolean} props.opened - State of modal being open or not
+ *  @param {Function} props.close - Function to close the modal
+ */
+export default function HospitalModel ({ opened, close }) {
+  const form = useForm({
+    mode: 'uncontrolled',
+    initialValues: {
+      name: '',
+      email: '',
+      phone: '',
+      address: '',
+    },
+
+    validate: {
+      name: (value) =>
+        /^[a-zA-Z\s\-']{2,30}$/.test(value)
+          ? null
+          : "Name must be between 2 and 30 characters long. It can only contain alphabetic characters (A-Z, a-z), spaces, hyphens (-), and apostrophes (').",
+      email: (value) => (/^\S+@\S+$/.test(value) ? null : 'Invalid email'),
+    },
+  });
+
+  /**
+   * Calls API to create an Hospital
+   * @param {object} props - The props that contain form values
+   * @param {string} props.role - Role value on the form
+   * @param {string} props.name - Name of recipient
+   * @param {string} props.email - Email of recipient
+   */
+  function onSubmit (values) {
+    fetch('/api/v1/hospitals', {
```

Comment:
I know you were following the previously implemented InviteModal, which is older code, but I think there's an opportunity to refactor/improve on that code here...

This creation is calling fetch directly, which means that the createHospital function in the LifelineAPI you added is not being used.

Perhaps we should rewrite this so that it uses a react-query mutation and the LifelineAPI like in other places?